loki/transform/transform_extract_contained_procedures.py

```diff
@@ -0,0 +1,198 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from loki.subroutine import Subroutine
+from loki.expression import (
+    FindVariables, FindInlineCalls, SubstituteExpressions,
+    DeferredTypeSymbol, Array, IntLiteral
+)
+from loki.ir import (
+    CallStatement, DerivedType
+)
+from loki.visitors import (
+    Transformer, FindNodes,
+)
+
```

Comment:
To explicitly list symbols that should be imported from this module, please add
```python
__all__ = ['extract_contained_procedures', 'extract_contained_procedure']
```

Reply:
Added this.